core/ecschema-editing/src/Editing/Editor.ts

```diff
@@ -61,9 +66,41 @@ export class SchemaContextEditor {
   public get schemaContext(): SchemaContext { return this._schemaContext; }
 
   public async finish(): Promise<SchemaContext> {
+    this._currentChanges = [];
     return this._schemaContext;
   }
 
+  public addEditInfo(changeInfo: ISchemaEditChangeInfo) {
+    const length = this._currentChanges.push(changeInfo);
+    changeInfo.sequence = length - 1;
+  }
+
+  public async revertCurrentChanges(): Promise<void> {
+    for (const change of this._currentChanges) {
+      await this.revertChange(change);
+    }
+
+    this._currentChanges = [];
+  }
+
+  public async revertChange(change: ISchemaEditChangeInfo): Promise<void> {
+    await change.revertChange();
+    this.removeChange(change);
+  }
+
+  public changeCancelled(change: ISchemaEditChangeInfo) {
+    this.removeChange(change);
+  }
+
+  public removeChange(change: ISchemaEditChangeInfo) {
+    this._currentChanges = this._currentChanges.splice(change.sequence, 1);
```

Comment:
I think that won't work. If I add 3 changes, the list would look like this (mocking up)...
```ts
0: { sequence: 0, editType: "SetLabel", ... }
1: { sequence: 1, editType: "SetPriority", ... }
2: { sequence: 2, editType: "SetDescription", ... }
```
If I now remove, the 2n change, and since all methods are public, we might want to have that possibility, it would get sliced out from the list
```ts
0: { sequence: 0, editType: "SetLabel", ... }
1: { sequence: 2, editType: "SetDescription", ... }
```
As the sequence of all other edits remain, if you want to call removeChange with the SetDescription edit, you'd call splice to remove the entry on index 2 (as that is the sequence value), but actually want to remove index 1.

So I guess we either want to prevent that changes can be removed anytime, or we need a lookup before calling splice like:
```ts
const changeIndex = this._currentChanges.find((entry) => entry.sequence === change.sequence)); // alternativly this._currentChanges.indexOf(change)
this._currentChanges = this._currentChanges.splice(changeIndex, 1);
```

Reply:
I removed the sequence property entirely.